app/forms/idv/finance_form.rb

```diff
@@ -39,6 +39,13 @@ def finance_type_valid
       end
     end
 
+    def ccn_valid
+      return true unless finance_type == :ccn
+      unless finance_account =~ /^\d{8}$/
+        errors.add :finance_account, I18n.t('idv.errors.invalid_ccn')
+      end
+    end
```

Comment:
My bad for not mentioning the message option. I believe you can pass it in like this:

```ruby
validates :finance_account, 
          format: { with: /^\d{8}$/, message: I18n.t('idv.errors.invalid_ccn') }, 
          if: :ccn?
```